src/offsetPath.js

```diff
@@ -47,12 +76,27 @@
     if (start.type !== 'ray' || end.type !== 'ray') {
       return internalScope.flip(serializeParsed(start), serializeParsed(end));
     }
+    if (start.size !== end.size) {
+      return internalScope.flip(serializeParsed(start), serializeParsed(end));
+    }
+    if (start.contain !== end.contain) {
+      return internalScope.flip(serializeParsed(start), serializeParsed(end));
+    }
 
     return {
       start: start.input,
       end: end.input,
       serialize: function (input) {
-        return 'ray(' + input + 'deg)';
+        var result = 'ray(' + input + 'deg';
+
+        if (start.size !== 'none') {
+          result += ' ' + start.size;
+        }
+        if (start.contain) {
+          result += ' contain';
+        }
+        result += ')';
+        return result;
```

Comment:
Avoid duplicating the logic here, just call the above method with the non-interpolable parts passed in in the argument.